src/robotFactory.js

```diff
@@ -1,10 +1,73 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = {};
```

Comment:
```suggestion
    this.coords = {
      x: coords.x ? coords.x : 0,
      y: coords.y ? coords.y : 0
    };
```
Why won't you assign these values right here?

Reply:
Totally agree with you, it will be really more obvious